lib/orocos/process.rb

```diff
@@ -974,13 +987,16 @@ def command_line(working_directory: Orocos.default_working_directory,
         #   true will enable valgrind support. Setting it to an array of strings will
         #   specify a list of arguments that should be passed to valgrind
         #   itself. This is obviously incompatible with the gdb option.
-        def spawn(log_level: nil, working_directory: Orocos.default_working_directory,
-                  cmdline_args: Hash.new,
-                  oro_logfile:  "orocos.%m-%p.txt",
-                  prefix: nil, tracing: Orocos.tracing?, name_service: Orocos::CORBA.name_service,
-                  wait: nil,
-                  output: nil,
-                  gdb: nil, valgrind: nil)
+        def spawn(
+            log_level: nil, working_directory: Orocos.default_working_directory,
+            cmdline_args: Hash.new,
+            oro_logfile:  "orocos.%m-%p.txt",
+            prefix: nil, tracing: Orocos.tracing?,
+            wait: nil,
+            output: nil,
+            gdb: nil, valgrind: nil,
+            ior_read: nil, ior_write: nil
```

Comment:
Do the pipe management here instead of doing it in Orocos.run. It would simplify everything by a lot.

Plus, currently, we can't call wait_running externally, which is something Syskit needs.

Reply:
What about https://github.com/rock-core/tools-orocosrb/blob/83c4369cfc0d9097ff2cc492c59a0a0c674b3569/lib/orocos/process.rb#L826?

Reply:
What about it ?

What I mean is: create the pipe and store it in the Process object. Since the pipe is in the process object, `#wait_running` will know about it ...